spec/archive_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "spec_helper"
+
+describe Alephant::Broker::LoadStrategy::S3::Archived do
+  subject { described_class.new }
+
+  describe "#s3_path" do
+    let(:id) { 42 }
+    let(:component_meta) { double(:id => id) }
+
+    specify do
+      expect(subject.s3_path component_meta).to eq id
+    end
+
+    context "no location associated with component meta" do
+      let(:component_meta) { double(:id => nil) }
+
+      specify do
+        expect do
+          subject.s3_path component_meta
+        end.to raise_error Alephant::Broker::InvalidCacheKey
+      end
```

Comment:
@stevenjack I personally subscribe to the way of thinking that the chosen answer here has: http://stackoverflow.com/questions/5170323/does-using-curly-braces-go-against-the-ruby-way?answertab=votes#tab-top

I like `do`/`end` as a marker to indicate where side-effects are - assists in designing with functional principles.  Not a deal-breaking issue for me personally (but that's my rationale anyway, in case there's any new thoughts in that for you)